google/cloud/bigtable/examples/opencensus/bigtable_opencensus.cc

```diff
@@ -0,0 +1,166 @@
+      instance_id);
+  //! [connect admin]
+
+  //! [create table]
+  // Define the desired schema for the Table.
+  auto gc_rule = google::cloud::bigtable::GcRule::MaxNumVersions(1);
+  google::cloud::bigtable::TableConfig schema({{"family", gc_rule}}, {});
+
+  // Create a table.
+  auto returned_schema = table_admin.CreateTable(table_id, schema);
+  //! [create table]
+
+  // Create an object to access the Cloud Bigtable Data API.
+  //! [connect data]
+  google::cloud::bigtable::Table table(
+      google::cloud::bigtable::CreateDefaultDataClient(
+          project_id, instance_id, google::cloud::bigtable::ClientOptions()),
+      table_id);
+  //! [connect data]
+
+  // Modify (and create if necessary) a row.
+  //! [write rows]
+  std::vector<std::string> greetings{"Hello World!", "Hello Cloud Bigtable!",
+                                     "Hello C++!"};
+  int i = 0;
+  for (auto const& greeting : greetings) {
+    // Each row has a unique row key.
+    //
+    // Note: This example uses sequential numeric IDs for simplicity, but
+    // this can result in poor performance in a production application.
+    // Since rows are stored in sorted order by key, sequential keys can
+    // result in poor distribution of operations across nodes.
+    //
+    // For more information about how to design a Bigtable schema for the
+    // best performance, see the documentation:
+    //
+    //     https://cloud.google.com/bigtable/docs/schema-design
+    std::string row_key = "key-" + std::to_string(i);
+    table.Apply(google::cloud::bigtable::SingleRowMutation(
+        std::move(row_key),
+        google::cloud::bigtable::SetCell("family", "c0", greeting)));
+    ++i;
+  }
+  //! [write rows]
+
+  // Read a single row.
+  //! [read row]
+  auto result = table.ReadRow(
+      "key-0",
+      google::cloud::bigtable::Filter::ColumnRangeClosed("family", "c0", "c0"));
+  if (not result.first) {
+    std::cout << "Cannot find row 'key-0' in the table: " << table.table_name()
+              << std::endl;
+    return 0;
+  }
+  auto const& cell = result.second.cells().front();
+  std::cout << cell.family_name() << ":" << cell.column_qualifier() << "    @ "
+            << cell.timestamp().count() << "us\n"
+            << '"' << cell.value() << '"' << std::endl;
+  //! [read row]
+
+  // Read a single row.
+  //! [scan all]
+  for (auto& row :
+       table.ReadRows(google::cloud::bigtable::RowRange::InfiniteRange(),
+                      google::cloud::bigtable::Filter::PassAllFilter())) {
+    std::cout << row.row_key() << ":\n";
+    for (auto& cell : row.cells()) {
+      std::cout << "\t" << cell.family_name() << ":" << cell.column_qualifier()
+                << "    @ " << cell.timestamp().count() << "us\n"
+                << "\t\"" << cell.value() << '"' << std::endl;
+    }
+  }
+  //! [scan all]
+
+  // Delete the table
+  //! [delete table]
+  table_admin.DeleteTable(table_id);
+  //! [delete table]
+
+#ifndef OPENCENSUS_DISABLED
```

Comment:
Is there a more deterministic way to terminate the reporting?  Maybe some OpenCensus API to query how many reports are pending?

Reply:
I couldn't figure out any API that could terminate gracefully ([code](https://github.com/census-instrumentation/opencensus-cpp)).  
I even checked [java example](https://opencensus.io/quickstart/java/tracing/) but couldn't find such API.

Reply:
Ack.